### PR TITLE
Address authentication failure cases

### LINKED ISSUE #160: Recommend that the AS keep members' evaluations consistent
@kkohbrok noted in https://github.com/mlswg/mls-protocol/pull/824#pullrequestreview-1213165509 that an important function of the AS is to ensure that clients come to consistent evaluations of credentials.  We should make that recommendation in this document, including the fact that consistency is needed when validity can change over time.

### PR DESCRIPTION
**Pull request by bifurcation**
Fixes #160

### CODE REVIEW

File: draft-ietf-mls-architecture.md [hunk 1]
```diff
@@ -765,6 +765,34 @@ proposals, but any other policies must be assured to be consistent as above.
...
+## Handling Authentication Failures
+
+Within an MLS group, every member is authenticated to other member, by means of
+credentials issued and verified by the Authentication Service.  MLS does not
+prescribe what actions, if any, an application should take in the event that a
+group member presents an invalid credential.  For example, an application may
+require such a member to be immediately evicted, or may allow some grace period
+for the problem to be remediated. To avoid operational problems, it is important
+for all clients in a group to have a consistent view of which credentials in a
+group are valid, and how to respond to invalidity.
```
Comment (rohan-wire) [inline draft-ietf-mls-architecture.md:777]: s/invalidity/invalid credentials/

**Review by rohan-wire — APPROVED** [review]

File: draft-ietf-mls-architecture.md (hunk 1 above)
Comment (bifurcation) [inline draft-ietf-mls-architecture.md:777]:
```suggestion
group are valid, and how to respond to invalid credentials.
```

File: draft-ietf-mls-architecture.md [hunk 2]
```diff
@@ -765,6 +765,34 @@ proposals, but any other policies must be assured to be consistent as above.
 > Have an explicit group policy setting the conditions under which external
 > joins are allowed.
 
+## Handling Authentication Failures
+
+Within an MLS group, every member is authenticated to other member, by means of
```
Comment (ekr) [inline draft-ietf-mls-architecture.md:770]:
```suggestion
Within an MLS group, every member is authenticated to other member by means of
```

File: draft-ietf-mls-architecture.md [hunk 3]
```diff
@@ -765,6 +765,34 @@ proposals, but any other policies must be assured to be consistent as above.
...
+credentials issued and verified by the Authentication Service.  MLS does not
+prescribe what actions, if any, an application should take in the event that a
+group member presents an invalid credential.  For example, an application may
+require such a member to be immediately evicted, or may allow some grace period
+for the problem to be remediated. To avoid operational problems, it is important
```
Comment (ekr) [inline draft-ietf-mls-architecture.md:775]: Is this supposed to be subsequent to group establishment? I would assume that at formation time, invalid users would simply be rejected.

File: draft-ietf-mls-architecture.md [hunk 4]
```diff
@@ -765,6 +765,34 @@ proposals, but any other policies must be assured to be consistent as above.
...
+> Have a uniform credential validation process to ensure that all group members
+> evaluate other members' credentials in the same way.
+
+> **RECOMMENDATION:**
+> Have a uniform policy for how invalid credentials are handled.
+
+In some authentication systems, it is possible for a previously-valid credential
+to become invalid over time.  For example, in a system based on X.509
+certificates, credentials can expire or be revoked.  Fortunately, the MLS update
+mechansisms allow a client to replace an old credential with a new one. This is
```
Comment (ekr) [inline draft-ietf-mls-architecture.md:789]:
```suggestion
certificates, credentials can expire or be revoked.  The MLS update
mechanisms allow a client to replace an old credential with a new one. This is
```

**Review by ekr — CHANGES_REQUESTED** [review]

File: draft-ietf-mls-architecture.md [hunk 5]
```diff
@@ -765,6 +765,34 @@ proposals, but any other policies must be assured to be consistent as above.
 > Have an explicit group policy setting the conditions under which external
 > joins are allowed.
 
+## Handling Authentication Failures
+
+Within an MLS group, every member is authenticated to other member, by means of
+credentials issued and verified by the Authentication Service.  MLS does not
```
Comment (beurdouche) [inline draft-ietf-mls-architecture.md:771]:
```suggestion
credentials verified by the Authentication Service.  MLS does not
```

File: draft-ietf-mls-architecture.md (hunk 3 above)
Comment (beurdouche) [inline draft-ietf-mls-architecture.md:775]:
```suggestion
for the problem to be remediated. To avoid operational problems, it is preferable
```

Comment (beurdouche) [reply draft-ietf-mls-architecture.md:775]: This is for external joins

**Review by beurdouche — APPROVED** [review]

**Pull request merged**